java-symbol-solver-model/src/main/java/me/tomassetti/symbolsolver/model/typesystem/ReferenceTypeUsage.java

```diff
@@ -292,7 +292,8 @@ protected boolean compareConsideringTypeParameters(ReferenceTypeUsage other) {
         }
         if (this.getQualifiedName().equals(other.getQualifiedName())) {
             if (this.parameters().size() != other.parameters().size()) {
-                throw new IllegalStateException();
+                // one have the type not specified so they are compatible
+                return true;
```

Comment:
Do you already know when you plan on merging this branch? I'd really like to help fixing those unsolved methods :)